challengeOneWeek3.js

```diff
@@ -1,15 +1,29 @@
 
-    var i;
-    var str='1ab2c3defUIoAu1806';
-    var even =[];
-    var odd =[];
+var str='1ab2c3defUIoAu18x0p6o9i20k100';
+// not yet sure if these should be global
     var numbers = str.match(/\d+/g)
     const chars = str.toLowerCase().match(/[A-z]/g).sort()
-    var sortedStr=[];
-    for (i=0; i<str.length;i++){
 
+    function evenAndOddNumbers(num){
+        var i;
+        var even =[];
+        var odd =[];
+        var isGreaterThan = function(a, b){return a-b}
+        for(i=0; i<num.length; i++){
+            if(num[i]%2===0){
+                even.push(num[i])
+            }
+            else if(num[i]%2 ===1){
+                odd.push(num[i]);
+           
+        }
+        }
+        
+    return object={
+        even:even.sort(isGreaterThan),
+        odd:odd.sort(isGreaterThan),
+        chars:chars
```

Comment:
Always avoid using global variables. 
Infact the function tells us that it returns even and odd numbers but we do have characters here.
One simple way to deal with this is to move the logic below to the `evenAndOddNumbers` function and change its name.
```js
 var numbers = str.match(/\d+/g)
 const chars = str.toLowerCase().match(/[A-z]/g).sort()	    
```